core/src/main/java/org/frankframework/jdbc/JdbcTransactionalStorage.java

```diff
@@ -566,122 +542,64 @@ protected String storeMessageInDatabase(Connection conn, String messageId, Strin
 				stmt.setTimestamp(++parPos, null);
 			}
 
-			if (!isStoreFullMessage()) {
-				if (isOnlyStoreWhenMessageIdUnique()) {
-					JdbcUtil.setParameter(stmt, ++parPos, messageId, getDbmsSupport().isParameterTypeMatchRequired());
-					stmt.setString(++parPos, getSlotId());
-				}
-				stmt.execute();
-				return null;
-			}
-			if (!dbmsSupport.mustInsertEmptyBlobBeforeData()) {
+			if (isStoreFullMessage()) {
 				int blobColumnIndex = ++parPos;
 				Object blobHandle = dbmsSupport.getBlobHandle(stmt, blobColumnIndex);
 				try (ObjectOutputStream oos = new ObjectOutputStream(getBlobOutputStream(dbmsSupport, blobHandle, stmt, blobColumnIndex, isBlobsCompressed()))) {
 					oos.writeObject(message);
 				}
 				dbmsSupport.applyBlobParameter(stmt, blobColumnIndex, blobHandle);
```

Comment:
This complicated code for inserting BLOB values is unfortunately something I don't dare to change yet, mostly for PostgreSQL.
Although for BLOB columns we might not need this anymore, even with PostgreSQL. However for CLOB columns this kludgy code pattern is still needed and we might as well treat BLOBs and CLOBs the same everywhere in our code.